app/views/halls/_form.html.erb

```diff
@@ -0,0 +1,9 @@
+<%= form_with model: @hall do |form| %>
+  <% @hall.errors.each do |error| %>
+    <p><%= error.full_message %></p>
+  <% end %>
+  <p><%= form.text_field :name, placeholder: 'name' %>
+  <%= form.text_field :capacity, placeholder: 'capacity' %>
+  <%= form.submit %>
+  <%= link_to "Cancel", @hall_path %></p>
```

Comment:
I would move "cancel" link outside the form. Then there would be also no need to define @hall_path in controller. You could simply add links in the new and edit views.

Reply:
I agree controller wasn't a good place for defining @hall_path, however I would rather not move the cancel link outside the form
Solution I chose was to change `<%= link_to "Cancel", cancel_path %>`, then define cancel_path as `<%= render "form", cancel_path: halls_path %>` and `<%= render "form", cancel_path: hall_path(@hall) %>` in new and edit views respectively